### ISSUE TITLE
Query

### ISSUE THREAD

**Issue by ali4321143** [rohinish404/image_captioning#1]
Salam/Good Evening
thanks for sharing the code
I have some question if you can answer
Q1 while training the LSTM in batches we have to pad so that all same same length in the batch and in order to not compute the loss on the pad token we use the ignore_idx in the cross entropy loss
Q2 we feed the start token at timestamp-0 but we do not feed the end token as input to the LSTM
Q3 while computing the loss we eliminate the start token and add the end token in ground truth then compute the loss with the predicted prob  model give it
Q4 in validation phase we use the AUTOREGRESSIVE technique not the teacher forcing
Q5 in the validation phase we stopped the seq when it generates the end token  so it might possible that generates token/prob are less than the tokens we have in the ground truth so DO we only compute the loss against the tokens equal to the tokens model gives the prob and rest tokens of ground truth are ignored from the loss?
